Automattic/wp-calypso
# Fix translations loading for Jetpack plans

#### Changes proposed in this Pull Request

Refactor the retrieval of plans array to account for situations where the translations for all the plans become available later than the translation for the current plan name. Originally addressed in https://github.com/Automattic/wp-calypso/pull/52629

#### Testing instructions



* Switch to a non-English user locale - non-Latin character set would show the issue more clearly.
* Have a linked Jetpack site, navigate to https://wordpress.com/plans/your-jetpack-site-url, reload the browser tab a few times until 'Security Daily', 'Complete' and 'Security Real-Time' display English titles and descriptions.
* Checkout the branch locally and start with `BUILD_TRANSLATION_CHUNKS=true ENABLE_FEATURES=use-translation-chunks,wpcom-user-bootstrap yarn start`
* Navigate to http://calypso.localhost:3000/plans/your-jetpack-site-url and confirm that all the plans are being translated now, even after multiple refreshes.

Related to #52629
